pkg/cli/cli.go

```diff
@@ -306,3 +310,75 @@ func RunPiped(ctx context.Context, cmd1, cmd2 CommandType, pids *PipedCommandsPi
 
 	return out.Bytes(), nil
 }
+
+// CheckAndGunzipFile checks if a file is gzipped and extracts it in that case...
+// ... it also returns the name of the unzipped file
+func CheckAndGunzipFile(fileName string) (string, error) {
+	file, err := os.Open(fileName)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		fileErr := file.Close()
+		if fileErr != nil {
+			log.Error(fileErr)
+		}
+	}()
+
+	// read first 512 bytes for http.DetectContentType
+	headerBytes := make([]byte, 512)
+	_, err = file.Read(headerBytes)
+	if err != nil {
+		return "", err
+	}
+
+	// check if file is gzipped
+	fileType := http.DetectContentType(headerBytes)
+	if fileType == gzipType {
```

Comment:
In order to avoid nesting, I'd change the order of this `if`..  

```go
if fileType != gzipType {
    return fileName, nil
}

// continue with gzip-code here
```